### PR TITLE
[ML] Trigger & control builds via comments and labels

### PR DESCRIPTION
**Pull request by edsavage**
Investigate behaviour of BuildKite's PR webhook.

Full testing won't be possible until `.buildkite/pull-requests.json` is merged to `main`

### CODE REVIEW

**Comment by edsavage** [comment]
buildkite  debug  on windows

**Comment by edsavage** [comment]
buildkite debug on linux

**Comment by edsavage** [comment]
buildkite debug on linux

**Comment by edsavage** [comment]
buildkite build this

**Comment by droberts195** [comment]
Please merge `main` into this PR branch after #2445 is merged so that it's clearer what's extra in this PR.

File: .buildkite/pipelines/build_linux.json.py [hunk 1]
```diff
@@ -141,6 +144,15 @@ def main(args):
                         choices=actions,
                         default="build",
                         help="Specify a build action.")
+    parser.add_argument("--snapshot",
+                        required=False,
+                        choices=build_snapshot,
+                        default=None,
+                        help="Specify if a snapshot build is wanted.")
+    parser.add_argument("--candidate",
```
Comment (droberts195) [inline .buildkite/pipelines/build_linux.json.py:152]: I think `candidate` should be `version_qualifier` throughout this PR. Using different terminology to the release manager, other repos, and some other scripts in this repo is only going to cause confusion in the future.

File: .buildkite/pipelines/build_windows.json.py [hunk 2]
```diff
@@ -99,6 +101,15 @@ def main(args):
...
                         help="Specify a build action.")
+    parser.add_argument("--snapshot",
+                        required=False,
+                        choices=build_snapshot,
+                        default=None,
+                        help="Specify if a snapshot build is wanted.")
+    parser.add_argument("--candidate",
+                        required=False,
+                        default=None,
+                        help="Specify a build candidate string.")
```
Comment (droberts195) [inline .buildkite/pipelines/build_windows.json.py:112]:
```suggestion
                        help="Specify a version qualifier.")
```

"Build candidate" is the wrong term. We have build candidates even for versions that have no qualifier.

**Review by droberts195 — APPROVED** [review]
LGTM

**Pull request merged**